src/libmongoc/doc/learn/get/from-source.rst

```diff
@@ -0,0 +1,386 @@
+
+  A new stable release of CMake can be obtained from
+  `the CMake downloads page`__.
+
+  __ https://cmake.org/download/#latest
+
+
+  For Windows and macOS, simply download the CMake `.msi`/`.dmg` (not the
+  `.zip`/`.tar.gz`) and use it to install CMake.
+
+  On Linux, download the self-extracting shell script (ending with `.sh`) and
+  execute it using the `sh` utility, passing the appropriate arguments to
+  perform the install. For example, with the CMake 3.27.0 on the `x86_64`
+  platform, the following command can be used on the
+  `cmake-3.27.0-linux-x86_64.sh` script::
+
+    $ sh cmake-3.27.0-linux-x86_64.sh --prefix="$HOME/.local" --exclude-subdir --skip-license
+
+  Assuming that `$HOME/.local/bin` is on your `$PATH` list, the `cmake` command
+  for 3.27.0 will then become available.
+
+  The `--help` option can be passed to the shell script for more information.
+
+For the remainder of this page, it will be assumed that `cmake` is available as
+a command on your `PATH` environment variable and can be executed as "`cmake`"
+from a shell. You can test this by requesting the `--version` from CMake from
+the command line::
+
+  $ cmake --version
+  cmake version 3.21.4
+
+  CMake suite maintained and supported by Kitware (kitware.com/cmake).
+
+.. important::
+
+  A CMake of version 3.15 *or newer* is **required** for building
+  |mongo-c-driver| from source.
+
+.. note::
+
+  If you intend to build |libbson| *only*, then CMake is sufficient for the
+  build. Additional C driver features may require additional external
+  dependencies be installed, but we will not worry about them here.
+
+.. XXX: Additional note for installing dependencies
+
+  The Linux dependencies of sufficient version are likely available using the
+  system package manager. For example, on **Debian**/**Ubuntu** based systems,
+  they can be installed using APT::
+
+    # apt install libssl-dev libsasl2-dev
+
+  On **RedHat** based systems (**Fedora**, **CentOS**, **RockyLinux**,
+  **AlmaLinux**, etc.)::
+
+    # dnf install openssl-devel cyrus-sasl-devel
+
+  Package names may vary between distributions.
+
+
+.. _learn.get.configure:
+
+Configuring for |libbson|
+*************************
+
+.. important::
+
+  If you are building with Xcode [#xcode_env]_ or Visual Studio [#vs_env]_, you
+  may need to execute CMake from within a special environment in which the
+  respective toolchain is available.
+
+Let the name `$BUILD` be the path `$SOURCE/_build`. This will be the directory
+where our built files will be written by CMake.
+
+With the source directory for |mongo-c-driver| at `$SOURCE` and build directory
+`$BUILD`, the following command can be executed from a command-line to configure
+the project with both |libbson| and |libmongoc|::
+
+  $ cmake -S $SOURCE -B $BUILD \
+    -D ENABLE_EXTRA_ALIGNMENT=FALSE \
+    -D ENABLE_AUTOMATIC_INIT_AND_CLEANUP=FALSE \
+    -D CMAKE_BUILD_TYPE=RelWithDebInfo \
+    -D ENABLE_MONGOC=FALSE
```

Comment:
```suggestion
    -D ENABLE_MONGOC=OFF
```

In 1.24.3, this results in an error:
```
CMake Error at CMakeLists.txt:162 (message):
  ENABLE_MONGOC option must be ON or OFF
```
I expect this will not be an error in 1.25.0. Building older versions may encounter this error. Suggestion applies to other references of `ENABLE_MONGOC`.